Web/src/components/room/Meta.ts

```diff
@@ -0,0 +1,30 @@
+import { Presence } from 'phoenix';
+
+
+export interface HostMeta extends Meta {
+  isHost: true,
+  game: string,
+}
+
+export interface NonHostMeta extends Meta {
+  isHost: false,
+}
+
+export interface Meta {
+  isHost: boolean,
+  name: string,
+  game?: string,
+}
+
+/**
+ * Returns tuples of client id and Meta
+ */
+export function getMetas(presence: Presence): [string, Meta][] {
+  return presence.list((id, { metas }) => [id, metas[0]]);
+}
+
+export function findHostMeta(metas: [string, Meta][]): [string, HostMeta] | null {
+  const maybeHostMeta = metas.find(([_, meta]) => meta.isHost);
```

Comment:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

will this work and will it remove the need for typecasting and `if (null) return null`

Reply:
ok nvm i realised it wont work